pkg/server/api/svid/v1/server_test.go

```diff
@@ -2,35 +2,201 @@ package svid_test
+		serviceErr string
+		spiffeID   spiffeid.ID
+	}{
+		{
+			name: "success",
+			certChain: [][]byte{
+				[]byte(rawCert),
+			},
+			expiresAt: time.Now().Add(time.Minute),
+			req: &svidpb.MintX509SVIDRequest{
+				Csr: csrRaw,
+				Ttl: 10,
+			},
+			spiffeID: spiffeID,
+		}, {
+			name: "empty CSR",
+			err:  "request missing CSR",
+			code: codes.InvalidArgument,
+			msg:  "Request missing CSR",
+			req: &svidpb.MintX509SVIDRequest{
+				Csr: []byte{},
+				Ttl: 10,
+			},
+		}, {
+			name: "invalid CSR",
+			err:  "invalid CSR: asn1: structure error: tags don't match",
+			code: codes.InvalidArgument,
+			msg:  "Invalid CSR",
+			req: &svidpb.MintX509SVIDRequest{
+				Csr: []byte("invalid CSR"),
+				Ttl: 10,
+			},
+		}, {
+			name: "service return error",
+			err:  "some error",
+			code: codes.Internal,
+			msg:  "Invalid CSR",
+			req: &svidpb.MintX509SVIDRequest{
+				Csr: csrRaw,
+				Ttl: 10,
+			},
+			serviceErr: "some error",
+		},
 	}
 
-	conn, done := spiretest.NewAPIServer(t, registerFn, contextFn)
-	defer done()
+	for _, testCase := range testCases {
+		testCase := testCase
+		t.Run(testCase.name, func(t *testing.T) {
+			// Setup service
+			c.service.certChain = testCase.certChain
+			c.service.id = testCase.spiffeID
+			c.service.err = testCase.serviceErr
+			c.service.expiresAt = testCase.expiresAt
+
+			// Call mint
+			resp, err := c.svidClient.MintX509SVID(ctx, testCase.req)
 
-	client := svidpb.NewSVIDClient(conn)
+			// Verify expected error
+			if testCase.err != "" {
+				spiretest.RequireGRPCStatusContains(t, err, testCase.code, testCase.err)
+				require.Nil(t, resp)
+				require.Equal(t, testCase.msg, c.logHook.LastEntry().Message)
 
-	resp, err := client.MintX509SVID(context.Background(), &svidpb.MintX509SVIDRequest{})
-	spiretest.RequireGRPCStatus(t, err, codes.Unimplemented, "not implemented")
-	require.Nil(t, resp)
+				return
+			}
+
+			// Verify response
+			require.NoError(t, err)
+
+			require.Equal(t, testCase.spiffeID.TrustDomain().String(), resp.Svid.Id.TrustDomain)
+			require.Equal(t, testCase.spiffeID.Path(), resp.Svid.Id.Path)
+			require.Equal(t, testCase.expiresAt.UTC().Unix(), resp.Svid.ExpiresAt)
+			require.Equal(t, testCase.certChain, resp.Svid.CertChain)
+
+			c.logHook.Reset()
+		})
+	}
 }
 
 type FakeService struct {
 	svid.Service
+
+	id        spiffeid.ID
+	certChain [][]byte
+	err       string
+	expiresAt time.Time
+}
+
+func (s *FakeService) MintX509SVID(context.Context, *x509.CertificateRequest, time.Duration) (*api.X509SVID, error) {
+	if s.err != "" {
+		return nil, status.Errorf(codes.Internal, s.err)
+	}
+
+	var certs []*x509.Certificate
+	for _, cert := range s.certChain {
+		certs = append(certs, &x509.Certificate{Raw: cert})
+	}
+	return &api.X509SVID{ID: s.id, CertChain: certs, ExpiresAt: s.expiresAt}, nil
+}
+
+type testConfig struct {
+	svidClient svidpb.SVIDClient
+	service    *FakeService
+	logHook    *test.Hook
+	closeConn  func()
+}
+
+func (c *testConfig) close() {
```

Comment:
Even though the type is unexported, the export status of functions is a good indication about what methods on the unexported struct are "public" and not internal details to other code in the package.
```suggestion
func (c *testConfig) Close() {
```